nautobot/ipam/api/serializers.py

```diff
@@ -229,23 +226,14 @@ class Meta:
         extra_kwargs = {
             "family": {"read_only": True},
             "prefix_length": {"read_only": True},
+            "nat_outside_list": {"read_only": True},
         }
 
-    @extend_schema_field(str)
-    def get_nat_outside_list(self, obj):
-        try:
-            nat_outside_list = obj.nat_outside_list
-        except IPAddress.DoesNotExist:
-            return None
-        depth = get_nested_serializer_depth(self)
-        data = return_nested_serializer_data_based_on_depth(
-            IPAddressSerializer(nat_outside_list, context={"request": self.context.get("request")}),
-            depth,
-            obj,
-            nat_outside_list,
-            "nat_outside_list",
-        )
-        return data
+    def get_field_names(self, declared_fields, info):
+        """Add nat_outside_list reverse relation to the automatically discovered fields."""
+        fields = list(super().get_field_names(declared_fields, info))
+        self.extend_field_names(fields, "nat_outside_list")
+        return fields
```

Comment:
Might be a tricky one to lint around. This change works for MtoM (like `SecretsGroup.secrets_group_associations` above) and for Mto1 reverse relations (like `IPAddress.nat_outside_list` here)because DRF sees them as serializable fields, it just doesn't include such fields in `__all__` by default.

For GFKs and such I think we'll continue to need to handle them separately.